File: firebase/test/operation_reminder_copy.test.js
const test = require("node:test");
const assert = require("node:assert/strict");

const { buildTomorrowPrepCopy } = require("../functions/operation_reminder_copy");

test("buildTomorrowPrepCopy — no delivery", () => {
  const copy = buildTomorrowPrepCopy({ totalDelivery: 0, notStartedCount: 0 });
  assert.equal(copy.title, "Tomorrow's Preparation");
  assert.equal(copy.body, "No deliveries scheduled for tomorrow.");
  assert.equal(copy.deliveryCount, 0);
  assert.equal(copy.pendingCount, 0);
  assert.equal(copy.navTarget, "tomorrow_preparation");
});

test("buildTomorrowPrepCopy — one delivery pending", () => {
  const copy = buildTomorrowPrepCopy({ totalDelivery: 1, notStartedCount: 1 });
  assert.equal(copy.body, "1 delivery scheduled\n1 order pending preparation");
});

test("buildTomorrowPrepCopy — multiple deliveries pending", () => {
  const copy = buildTomorrowPrepCopy({ totalDelivery: 8, notStartedCount: 3 });
  assert.equal(
    copy.body,
    "8 deliveries scheduled\n3 orders pending preparation",
  );
});

test("buildTomorrowPrepCopy — all ready", () => {
  const copy = buildTomorrowPrepCopy({ totalDelivery: 8, notStartedCount: 0 });
  assert.equal(copy.body, "8 deliveries scheduled\nAll orders are ready.");
});

test("buildTomorrowPrepCopy — never uses programmer plural syntax", () => {
  const copy = buildTomorrowPrepCopy({ totalDelivery: 4, notStartedCount: 2 });
  assert.doesNotMatch(copy.body, /order\(s\)/);
  assert.doesNotMatch(copy.body, /not started/);
});

File: test/operation_reminder_copy_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:tfg_vday/backend/operation_reminder_copy.dart';

void main() {
  group('buildTomorrowPrepNoticeBody', () {
    test('no delivery', () {
      expect(
        buildTomorrowPrepNoticeBody(deliveryCount: 0, pendingCount: 0),
        'No deliveries scheduled for tomorrow.',
      );
    });

    test('one delivery pending', () {
      expect(
        buildTomorrowPrepNoticeBody(deliveryCount: 1, pendingCount: 1),
        '1 delivery scheduled\n1 order pending preparation',
      );
    });

    test('multiple deliveries pending', () {
      expect(
        buildTomorrowPrepNoticeBody(deliveryCount: 8, pendingCount: 3),
        '8 deliveries scheduled\n3 orders pending preparation',
      );
    });

    test('all ready', () {
      expect(
        buildTomorrowPrepNoticeBody(deliveryCount: 8, pendingCount: 0),
        '8 deliveries scheduled\nAll orders are ready.',
      );
    });
  });
}
